app/controllers/sessions_controller.rb

```diff
@@ -6,8 +6,14 @@ def create
     user = User.find_by(email: params[:email])
     if user.authenticate(params[:password])
       session[:user_id] = user.id
-      flash[:success] = "Welcome, #{user.name}!"
-      redirect_to "/users/#{user.id}"
+      flash[:success] = "Welcome, #{user.name}, you are logged in!"
+      if current_merchant?
```

Comment:
We could try something like this:

CASE
   WHEN current_merchant? THEN redirect_to "/merchant"
   WHEN current_admin? THEN redirect_to "/admin"
   WHEN .....
END

Not sure really how much more readable this makes it